src/wp-includes/fonts/class-wp-font-collection.php

```diff
@@ -0,0 +1,260 @@
+<?php
+/**
+ * Font Collection class.
+ *
+ * This file contains the Font Collection class definition.
+ *
+ * @package    WordPress
+ * @subpackage Fonts
+ * @since      6.5.0
+ */
+
+/**
+ * Font Collection class.
+ *
+ * @since 6.5.0
+ *
+ * @see wp_register_font_collection()
+ */
+final class WP_Font_Collection {
+	/**
+	 * The unique slug for the font collection.
+	 *
+	 * @since 6.5.0
+	 * @var string
+	 */
+	public $slug;
+
+	/**
+	 * Font collection data.
+	 *
+	 * @since 6.5.0
+	 * @var array|WP_Error|null
+	 */
+	private $data;
+
+	/**
+	 * Font collection JSON file path or URL.
+	 *
+	 * @since 6.5.0
+	 * @var string|null
+	 */
+	private $src;
+
+	/**
+	 * WP_Font_Collection constructor.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string       $slug         Font collection slug.
+	 * @param array|string $data_or_file Font collection data array or a path/URL to a JSON file
+	 *                                   containing the font collection.
+	 *                                   See {@see wp_register_font_collection()} for the supported fields.
+	 */
+	public function __construct( $slug, $data_or_file ) {
+		$this->slug = sanitize_title( $slug );
+
+		if ( is_array( $data_or_file ) ) {
+			$this->data = $this->sanitize_and_validate_data( $data_or_file );
+		} else {
+			// JSON data is lazy loaded by ::get_data().
+			$this->src = $data_or_file;
+		}
+
+		if ( $this->slug !== $slug ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Font collection slug. */
+				sprintf( __( 'Font collection slug "%s" is not valid. Slugs must use only alphanumeric characters, dashes, and underscores.' ), $slug ),
+				'6.5.0'
+			);
+		}
+	}
+
+	/**
+	 * Retrieves the font collection data.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array|WP_Error An array containing the font collection data, or a WP_Error on failure.
+	 */
+	public function get_data() {
+		// If the collection uses JSON data, load it and cache the data/error.
+		if ( $this->src && empty( $this->data ) ) {
+			$this->data = $this->load_from_json( $this->src );
+		}
+
+		if ( is_wp_error( $this->data ) ) {
+			return $this->data;
+		}
+
+		// Set defaults for optional properties.
+		$defaults = array(
+			'description' => '',
+			'categories'  => array(),
+		);
+		return wp_parse_args( $this->data, $defaults );
```

Comment:
```suggestion

		return wp_parse_args( $this->data, $defaults );
```

Empty line before the final `return`.

Reply:
Addressed in Gutenberg here: https://github.com/WordPress/gutenberg/pull/58691/commits/b27b0ad17903efae6c597233d2ae40441b72bd01

Reply:
Changed in https://github.com/WordPress/wordpress-develop/pull/6027/commits/095fd857413b24fd674284cd3229e4100cf0d994 ✅